source/Patches/EndGameSummary.cs

```diff
@@ -0,0 +1,141 @@
+                public string PlayerName { get; set; }
+                public Faction Faction { get; set; }
+                public Role CurrentRole { get; set; }
+                public Modifier Modifier { get; set; }
+                public int TasksCompleted { get; set; }
+                public int TasksTotal { get; set; }
+            }
+        }
+
+        public static void UpdatePlayerInfo()
+        {
+            Il2CppSystem.Collections.Generic.List<GameData.PlayerInfo> players = GameData.Instance.AllPlayers;
+
+            var pinfolist = new List<AdditionalTempData.PlayerInfo>();
+
+            foreach (GameData.PlayerInfo player in players)
+            {
+                byte pid = player.PlayerId;
+                string name = player.PlayerName;
+                Role role = Role.GetRole(player);
+                Modifier modifier = Modifier.GetModifier(player);
+                Faction faction = role.Faction;
+                int tasksdone = 0;
+                int taskstotal = 0;
+
+                AdditionalTempData.PlayerInfo pinfo = new AdditionalTempData.PlayerInfo();
+                pinfo.PlayerName = name;
+                pinfo.CurrentRole = role;
+                pinfo.Faction = faction;
+                pinfo.Modifier = modifier;
+
+                foreach (var task in player.Tasks)
+                {
+                        if (task.Complete == true)
+                        {
+                            tasksdone++;
+                        }
+                        taskstotal++;
+                }
+                
+                pinfo.TasksCompleted = tasksdone;
+                pinfo.TasksTotal = taskstotal;
+                pinfolist.Add(pinfo);
+            }
+            AdditionalTempData.playerData = pinfolist;
+        }
+
+        public static bool HasTasks(Faction faction)
+        {
+            return faction == Faction.Crewmates;
+        }
+
+        static bool IsNull(AdditionalTempData.PlayerInfo pinfo, string type)
+        {
+            return type switch {
+                "Role" => pinfo.CurrentRole == null,
+                "Mod" => pinfo.Modifier == null,
+                _ => true
+            };
+        }
+
+        public static void LoadGameSummary(EndGameManager __instance)
+        {
+            var position = Camera.main.ViewportToWorldPoint(new Vector3(0f, 1f, Camera.main.nearClipPlane));
+            GameObject roleSummary = UnityEngine.Object.Instantiate(__instance.WinText.gameObject);
+            roleSummary.transform.position = new Vector3(__instance.ExitButton.transform.position.x + 0.1f, position.y - 0.1f, -14f);
+            roleSummary.transform.localScale = new Vector3(1f, 1f, 1f);
+
+            var roleSummaryText = new StringBuilder();
+
+            roleSummaryText.AppendLine("End-of-game Summary:");
+
+            foreach (var data in AdditionalTempData.playerData)
+            {
+                var won = "   ";
```

Comment:
Suggestion: Java has a keyword `final`; any variable defined as `final` can only be written to exactly one. C# doesn't actually have a `final`, but I suggest you write code that would work that way. So instead of using a `won` string and rewrite it, define the variable once and write to it once in each branch. So something like:

```string won;
if (TempData.winners.Contains(w -> w.Name == data.PlayerName)
{
   won = won = $"[<color=#FAD934FF>W</color>]";
}
else {
   won = "   ";
}
```

This is a pretty benign example, but it's a good habit to get into because re-assigning variables can make it *very* hard to understand what a piece of code does later when you read it.

Reply:
i have changed it so the branches are more clear as suggested, couldent get the -> working with WinningPlayerData though as in your suggested code, so did it the same way i was already doing it.

Reply:
i have had to revert the changes i made for this as the changes i made seem to be causing the bug with the lovers not being shown as winners... i will put a comment in to reflect this.